### ISSUE TITLE
Errror  mkdir /host_mnt/c: file exists   when restarting docker container with mount

### ISSUE THREAD

**Issue by eino-makitalo** [docker/for-win#1560]
### Expected behavior
Container should start normally
### Actual behavior
C:\Users\einom>docker start f96263b10996
Error response from daemon: error while creating mount source path '/host_mnt/c/Users/einom/Documents/projects/cap/src': mkdir /host_mnt/c: file exists
Error: failed to start containers: f96263b10996
### Information

Docker version: 17.12.0-ce-win47 (15139)

Windows 10 Pro
Version: 1709
Build: 16299.192

Whole C drive is shared with docker VM

### Steps to reproduce the behavior

  1. start container with command similar than me in my home dir
docker run -d --name sonarqube -p 9000:9000 -p 9092:9092 -v c:/Users/einom/Documents/projects/cap/src:/src:rw sonarqube 
  2.  use container and enjoy that  directory in container /src is mounted OK
  3.  docker stop this container
  4.  docker start container ... and you got error  ... mkdir /host_mnt/c file exist.  

Of cource it exists because if I understand correctly this  /host_mnt/c should point to my whole C: drive ?

**Comment by eino-makitalo** [comment]
![image](https://user-images.githubusercontent.com/1835186/34957710-61a73628-fa37-11e7-8067-bd3af3b0513b.png)

**Comment by ibi8588** [comment]
I had the same issue. I was able to resolve it by running: 
docker volume rm -f [name of docker volume with error]
Then restarting docker, and running:
docker-compose up -d --build

**Comment by aalteirac** [comment]
I have the same issue, not even sure it's only after a container restart, seems like a "time-out" :-/

**Comment by ohernandez-getinsoft** [comment]
Got same error when trying to do `docker-compose run`

Here are my specs:
### Docker
Client:
 Version:       17.12.0-ce
 API version:   1.35
 Go version:    go1.9.2
 Git commit:    c97c6d6
 Built: Wed Dec 27 20:05:22 2017
 OS/Arch:       windows/amd64

Server:
 Engine:
  Version:      17.12.0-ce
  API version:  1.35 (minimum version 1.12)
  Go version:   go1.9.2
  Git commit:   c97c6d6
  Built:        Wed Dec 27 20:12:29 2017
  OS/Arch:      linux/amd64
  Experimental: true

### Docker-compose
docker-compose version 1.18.0, build 8dd22a96
docker-py version: 2.6.1
CPython version: 2.7.14
OpenSSL version: OpenSSL 1.0.2k  26 Jan 2017

### OS
Windows 10 Pro

I was just working on a dev-custom image with some volumes when suddenly i got a weird behaviour so i restarted the container and when tried to run that container the error happened

> $ docker-compose run --rm --service-ports web
Starting erpassetel_db_1 ... done
Error response from daemon: error while creating mount source path '/host_mnt/c/Users/randomUser/Documents/folder1/subFolder2/Addons': mkdir /host_mnt/c: file exists

After 5 minutes i just tried to run the containers again and it all worked well again

**Comment by marcel-dempers** [comment]
Got the same on windows:

```
 docker run -it -v "C:\git\somerepo\somefolder":/usr/lib/src alpine:3.7
docker: Error response from daemon: error while creating mount source path '/host_mnt/c/git/somerepo/somefolder': mkdir /host_mnt/c: file exists.
```

**Comment by rfulwell** [comment]
On Windows this may be due to a user password change. Uncheck the box to stop sharing the drive and then allow Docker to detect that you are trying to mount the drive and share it.
![image](https://user-images.githubusercontent.com/470472/36745728-49101172-1ba5-11e8-94da-1386cfba9702.png)

**Comment by 0xTanvir** [comment]
I was able to resolve it by running at Windows PowerShell (Admin).

**Comment by Luto73** [comment]
I do have a similar problem. Yesterday I created a container at D:\Users\myname\docker\test. When I try to restart the container using
`docker run -d -p 8008:80 -v D:/Users/myname/docker/test:/var/www/html php:apache`
I get:
`cd72fdb49e2d5067c0f5d4ff6ab0ae87c038db8b22492e280c9e80fe550e3499
docker: Error response from daemon: error while creating mount source path '/host_mnt/d/Users/myname/docker/test': mkdir /host_mnt/d: file exists.`

I tried to start the container with the normal command shell and with Windows Power Shell, neither works. Any ideas?

**Comment by chavolla** [comment]
Follow the instructions stated in the following page: 

[http://support.divio.com/local-development/docker/how-to-use-a-directory-outside-cusers-with-docker-toolbox-on-windowsdocker-for-windows](http://support.divio.com/local-development/docker/how-to-use-a-directory-outside-cusers-with-docker-toolbox-on-windowsdocker-for-windows)

**Comment by jefflill** [comment]
We see this a few times a week on **17.12.0-ce-win47 (15139)**.

Sometimes this is transient and the problem goes away after retrying running the container a few times.  Other times, restarting Docker helps.  Every week or two, we need to reset Docker back to factory defaults to get this working again.

This doesn't appear to be a drive sharing security issue.  The problem appears to be that the mount point already exists and Docker is trying to mount it again, and that's what's failing.

**Comment by Szauka** [comment]
@rfulwell Solution works! But it's easier to use the Reset Credentials... button.
![image](https://user-images.githubusercontent.com/33459309/38609289-96062c54-3d75-11e8-936d-ae59ff07811c.png)

**Comment by andzejsw** [comment]
I just rebuilt all images and it worked.

**Comment by glaux** [comment]
Resetting the credentials like @rfulwell and @Szauka worked even in a case where the credentials hadn't changed - docker had simply stopped working after a wakeup.

**Comment by chromygabor** [comment]
Same happens to me. It occurs only after a while, or maybe after returning from sleep. But Reseting credentials solves this problem. For a while...

**Comment by borgdrone7** [comment]
This is obviously a bug which should be resolved. We are forced to reset docker (or credentials) all the time, it is not acceptable solution. This is introduced with the latest version as I was using previous versions for months and this was never happening before.

**Comment by darinrandal** [comment]
None of the above solutions worked for me. The issue I had in this case was I had a separate user setup since I used a Microsoft account. That all worked fine and dandy but then Microsoft released an update which reset all those permissions and I ended up with this issue.

**Comment by LiamKarlMitchell** [comment]
```ERROR: for apache  Cannot start service apache: b"error while creating mount source path '/host_mnt/c/Users/megab/.composer': mkdir /host_mnt/c/Users/megab/.composer: permission denied"
```

Why does it need C drive? store the .composer in the directory im running ```docker-compose up -d``` from...

Ah maybe ~/.composer to ./.composer in the composer.yml.

I actually just created a blank file and it seems like it worked.

**Comment by yccheok** [comment]
I solve it by simply run this safe command

    docker volume prune

**Comment by AlanCarlyle** [comment]
@rfulwell's  solution and @Szauka's simplification seemed to work for me on my Windows 10 Pro.
Thanks everyone for helping with this... and best of luck finding a code fix

**Comment by nhudson76** [comment]
FWIW - PHP Storm's Docker integration can contribute to this.  
It's fairly anecdotal, but I can reproduce it.
I was working in a container, stopped that & started another; 
connected PHPS to daemon, pruned when done, 1st container no longer builds.

Steps to reproduce:
- Version 18.06.0-ce-win72 (19098)
- PHPS 2018.1.1
- [AzureAD account drive share / user workaround in place]

1. ~/projectA$ docker-compose up -d
2. ?? (do whatever. collect underpants.)
3. ~/projectA$ docker stop $(docker ps -a -q)
4. ~/projectB$ docker-compose up -d
5. PHPStorm > View > Tool Windows > Docker > [Connect] > projectB_xxx_1 > ENV Vars
6. ~/projectB$ docker-compose down (confirmed all containers removed)
7. ~/projectA$ docker-compose up -d
8. error while creating mount ... mkdir /host_mnt/c: file exists
9. PHPStorm > Docker > Disconnect
10. ~/projectA$ docker-compose up -d
Builds OK

Note I did not use any start/stop functionality in PHPS, I simply clicked 1 container to inspect the ENV & I probably left it 'focused' while stopping/starting.  UDP connection probably remained intact, I did not look for this in logs.


Log reports "SMB signature verification returned error = -13" 
and flip-flops between
"proxy >> GET /v1.24/containers/json?all=true\n"
and
"proxy >> GET /v1.25/containers/json?all=true\n"
so maybe some discrepancy there? idk

**Comment by niktekusho** [comment]
I just encountered this.

```sh
$ docker version
Client:
 Version:           18.06.0-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        0ffa825
 Built:             Wed Jul 18 19:05:28 2018
 OS/Arch:           windows/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.0-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       0ffa825
  Built:            Wed Jul 18 19:13:46 2018
  OS/Arch:          linux/amd64
  Experimental:     false
```

Since I was sure I did not change my user password nor did modify permissions on the shared drives, I simply tried to restart the docker daemon and it solved the issue. I'll report back if it happens again (just to make sure my workaround remains consistent I guess...)

**Comment by incyg** [comment]
I experienced this as well when using [Testcontainers](https://www.testcontainers.org/) for integration testing on a Windows 10 Pro machine `Version 1803 (OS Build 17134.228)`.

That system uses the same Docker version as @niktekusho.

I was able to recover from that problem as mentioned by other commenters here. Note that no changes were made to the Docker installation or its configuration which would somehow explain this problem to appear.

**Comment by yanxiaobo** [comment]
I fixed it by start  windows network for docker

**Comment by marcoebbinghaus** [comment]
Having the same bug. I created the container successfully (including a volume) and got the error message 

> Error response from daemon: error while creating mount source path '/host_mnt/c/Users/marcoebbinghaus/.gradle': mkdir /host_mnt/c: file exists

when trying to start it with docker start.

Had to remove / create the container three times and then I could start it successfully.

**Comment by CyberGirl007** [comment]
When I wanted to set up Windows Subsystem for Linux on my laptop, I had the following error:

```
Removing db_1
Recreating 7a6d58593003_db_1 ...
Recreating 7a6d58593003_db_1 ... error

ERROR: for 7a6d58593003_db_1  Cannot start service db: error while creating mount source path '/host_mnt/c/Users/user/project/dbdata': mkdir /host_mnt/c: file exists

ERROR: for db  Cannot start service db: error while creating mount source path '/host_mnt/c/Users/user/project/dbdata': mkdir /host_mnt/c: file exists
ERROR: Encountered errors while bringing up the project.
```

What worked for me was:

1. Remove all containers: `docker rm $(docker ps -aq)`
2. Remove all volumes that I can: `docker volume rm $(docker volume ls -q)`
3. Restarting Docker for Windows: right click on Docker in the tray, click Quit Docker, and restart Docker for Windows
4. Unsharing and re-sharing the C:/ drive in the Docker settings
5. Running `docker-compose up --build`.

**Comment by bwhitacrejha** [comment]
Resetting the credentials like @rfulwell and @Szauka worked for me as well even in a case where the credentials hadn't changed

**Comment by EkoMcCloud** [comment]
Same here, resetting credentials usually works when I change my password but I had this error message for 3 days in a row without a single password change.

**Comment by iamwwc** [comment]
Well, I use windows symbol link, because of golang's `$GOPATH`, I store my codes into `D:\\` but I link `D:\\codes` to my `$GOPATH/src/iamwwc.com/codes` so I can write codes in `$GOPATH`,  When I run `docker-compose up -d` in `$GOPATH/src/iamwwc.com/codes`, this error occur, when run in `D:\\codes`, all worked

SO please pay attention to whether you use symbol link

**Comment by remmeier** [comment]
I use git bash on windows a lot and frequently run into the issue. Paths are handled in there a bit differently as well (like /c/ instead of c:/). Not sure whether that might be related.

**Comment by pikenikes** [comment]
This issue happens to me when there's any public network on my machine. I can always get it fixed by setting network interfaces as private and restarting docker after that.

![image](https://user-images.githubusercontent.com/20577216/46570864-11a87c80-c963-11e8-922d-adb5368e14bf.png)

Works like a charm then:
![image](https://user-images.githubusercontent.com/20577216/46570871-2c7af100-c963-11e8-80f1-5ddd6fa34d56.png)

It seems to me that Docker machine looses permissions over file system when networks go public. Even though this fixes the issue, it isn't permanent. All it takes is a restart to get Windows setting networks public again and the procedure has to be repeated.

Hope this helps anyone facing the same issue.

**Comment by JonKoala** [comment]
@iamwwc's comment was on point with my case, i was having the same problem trying to start a container using a symlink.
It works using junctions though.

**Comment by hammypants** [comment]
Thanks to @Szauka as my issue was definitely a changed pw issue.

**Comment by LiamKarlMitchell** [comment]
Changed password issue, Reset credentials in Docker then docker-compose up resolved this.

```error while creating mount source path '/host_mnt/f/Dev/project': mkdir /host_mnt/f: file exists"```

**Comment by enhancednaong** [comment]
@rfulwell and @Szauka ' solutions worked for me! 
the reason was changing password periodically.

**Comment by obriat** [comment]
### Reset credential works ! 
Although I knew the solution, I never thought about it because my containers started flawlessly, it's only after 15 mn or so that files mounts broke suddenly... I don't know how docker could mount them at first without the correct credentials...

**Comment by remmeier** [comment]
for sure not a password issue in my case, so there is still a deeper issue somewhere i think. but restarting docker helped everytime so far

**Comment by obriat** [comment]
I still get the error frequently. To me the easiest workaround it to uncheck>apply>check>apply my "share drive" (without the need to restart docker).

**Comment by rfay** [comment]
We see this often on Windows test machines running tests. Regardless of any action with docker, it will typically work fine on the test retry.  IMO all the workarounds here are probably valid, but I think this is mostly just an intermittent problem.

**Comment by jefflill** [comment]
We also see this when running unit tests on Windows start start/stop lots of containers with mounted volumes but we also see this sometimes when starting containers manually too.

I suspect that this may be a Docker race condition where Docker may sometimes may not have fully removed an old volume mount before creating the new one.  This is actually fairly serious problem and I'm surprised that it hasn't been addressed yet since Windows is an important platform for Docker.  Perhaps this isn't an issue for Kubernettes mode or when running on Windows Server.

This thread is getting really long, so I'm going to submit a new issue and see what happens.  Unfortunately, I'm not a **golang** developer so I haven't tried to investigate this.

**Comment by kiahmed** [comment]
> I had the same issue. I was able to resolve it by running:
> docker volume rm -f [name of docker volume with error]
> Then restarting docker, and running:
> docker-compose up -d --build
> 
> I tried these same steps without restarting my docker, but restarting my computer and that didn't resolve the issue. What resolved the issue for me was removing the volume with the error, restarting my docker, then doing a build again.

what if removing volume isnt an option. This could happen if docker for windows auto updates docker and loses existing drive permission for some reason or somehow engine lost the credentials that you set first time to access the drive (i ma talking about windows scenario) . 

I resolved it by manually un ticking the drive access and tick it back with the credentials again.

**Comment by noglitchyo** [comment]
> On Windows this may be due to a user password change. Uncheck the box to stop sharing the drive and then allow Docker to detect that you are trying to mount the drive and share it.
> ![image](https://user-images.githubusercontent.com/470472/36745728-49101172-1ba5-11e8-94da-1386cfba9702.png)

Genius. I did not think about it and it was that. Thank you very much.

**Comment by sometumi** [comment]
docker volume rm -f $(docker volume ls -q)

**Comment by VitalyKrenel** [comment]
Why no one from docker development team has addressed this issue so far? The problem is quite annoying but it doesn't look like the hardest problem to solve (whether it is a race condition or just failed remounting attempt).
 
If there was a try solving this issue and complications emerged I would easily understand. But the complete silence when so many developers encountering the problem is unimaginable.

**Comment by u007** [comment]
yes this is quite a pain,
ive client that actually requires user to reset their password every 30days

**Comment by rfgamaral** [comment]
I think I'm having a problem where the cause is this same as this issue.

Take the following `docker-compose-yml` for instance:

```yaml
version: '3.5'
services:
  traefik:
    image: traefik:1.7-alpine
    volumes:
      - type: bind
        source: ./placeholder.txt
        target: /placeholder.txt
        volume:
          nocopy: true
    restart: always
```

Running `docker-compose up -d` works just fine (tried this on WSL and PowerShell), what doesn't work is the `restart: always`. I mean, say I restart Docker engine:

![image](https://user-images.githubusercontent.com/96476/51175028-1d380780-18b1-11e9-9ded-2453bbd8fe98.png)

I expect my container to start automatically but it doesn't:

```
docker ps --all
CONTAINER ID        IMAGE                COMMAND                  CREATED             STATUS                       PORTS               NAMES
a9488288d653        traefik:1.7-alpine   "/entrypoint.sh trae…"   4 minutes ago       Exited (255) 3 minutes ago   80/tcp              docker-test_traefik_1
```

**Comment by rfay** [comment]
I'd be interested to know if anybody has found a workaround. Our Windows test machines get this so very often:
```
C:\Program Files\Docker\Docker\Resources\bin\docker.exe: Error response from daemon: error while creating mount source path '/c/Users/testbot/tmp/buildkite/testbot-dell-win10pro-2-1/drud/quicksprint-windows': mkdir /c: file exists.
 ```

I'm thinking about wrapping docker-compose in a script that retries (it normally succeeds the next time, or the time after, same machine, no config change). But that seems so very ugly. But it would deal with the fact that we have to push the button to retry these tests.

**Comment by gimntut** [comment]
I changed the Windows user password.
Previously, the password was written in Cyrillic. I got the error in 2-3 hours.
Now the numeric password. The mount error only appears after 2 weeks.

**Comment by dlencer** [comment]
For those looking for a workaround for the time being:
The issue seems to be some kind of timeout for the credentials required to access your (shared) drive as someone noted before. The problem might be related to using a Windows domain user accounts, but I cannot check that, unfortunately. By un- and then re-sharing drive c: as advised in this thread, I was reliably able to bring my container (which just serves some code stored on a folder on drive c: of the host and is started via `docker-compose up`) back up each time I was confronted with the error message `[...] mkdir /host_mnt/c: file exists` upon trying to restart it. Just retrying did not help, un-/resharing was the only way.

**Yet, as long as I keep on restarting this container (via `docker-compose restart <mycontainer>`) frequently enough, that is at least every 10 minutes, everything seems fine so far, the error does not show up anymore (fingers crossed).** You can use some simple script for that purpose.

It does not help to keep restarting some other dummy-container which does nothing but having access to some other folder on the same drive of the host, at least my test case failed. Obviously, this workaround will not be helpful for all use-cases, but in my special case during development it is ok as I have to restart frequently anyways. It is definitely more comfortable for me than having to un-/reshare the drive all the time in the GUI.

This issue is a real showstopper for Windows-users, I have a hard time advertising the use of docker in my organization because of it, so I hope someone will be able to fix it soon.

**Comment by VitalyKrenel** [comment]
I've decided to try Docker Toolbox on my computer a month ago (or so), because my team mate have had a problem with starting our containers with Toolbox. We've fixed everything, but that not what's matter.

What's matter is that on Docker Toolbox there's no such issue with that `Errror mkdir /host_mnt/c: file exists` error message. Though Docker Toolbox is consider as a legacy solution - Docker Native works only with Education, Pro or Enterprise editions of Windows 10 on the computers with the Hyper V support, that actually is available only on the up-to-date hardware - it is actually a more stable, then the Native version.

Try researching whether the Docker Toolbox is appropriate for your development and use it instead of the Native version, if it's appropriate. I know, it's not a complete solution, but that is what we can stick for now, while Docker is no addressing this issue at all (I hope they are, but there's no any apparent sign of that).

**Comment by oldium** [comment]
I switched to the Edge channel on my Windows 10 1809 computer and the problem has gone - I have not seen the mkdir error so far (cca 2 days, few restarts of the container). I am now using Docker Desktop 2.0.1.0 (30090) build c8b6ca6/edge, Engine 18.09.1, Compose 1.23.2.

**Comment by dlencer** [comment]
I switched to edge as you recommended, but the same error showed up again.

**Comment by herbinator** [comment]
runnig docker run --rm -v c:/Users:/data alpine ls /data
C:\Program Files\Docker\Docker\Resources\bin\docker.exe: Error response from daemon: error while creating mount source path '/host_mnt/c/Users': mkdir /host_mnt/c: file exists.

getting same error. It looks i get it when connected to vpn. when disconnecting vpn it's fine

**Comment by VolkerH** [comment]
I experienced this problem when connected to a VPN using the CISCO Anyconnect Secure Mobility Client.
Once I quit the Cisco VPN I can mount the shares in the docker container as normal.

**Comment by herbinator** [comment]
what helps as workaround is changing the subnet adress in network settings and then it works fine again.

**Comment by JackSlaterIV** [comment]
> 
> 
> This issue happens to me when there's any public network on my machine. I can always get it fixed by setting network interfaces as private and restarting docker after that.
> 
> ![image](https://user-images.githubusercontent.com/20577216/46570864-11a87c80-c963-11e8-922d-adb5368e14bf.png)
> 
> Works like a charm then:
> ![image](https://user-images.githubusercontent.com/20577216/46570871-2c7af100-c963-11e8-80f1-5ddd6fa34d56.png)
> 
> It seems to me that Docker machine looses permissions over file system when networks go public. Even though this fixes the issue, it isn't permanent. All it takes is a restart to get Windows setting networks public again and the procedure has to be repeated.
> 
> Hope this helps anyone facing the same issue.

Thanks, did you find any way to make this a permanent workaround (survive to each Windows restart)?

**Comment by pikenikes** [comment]
JackSlaterIV, I actually have been able to make this workaround permanent, all it takes is defining a PowerShell script where you get all network interfaces and loop them setting as private. With this script, you can define a startup task on Windows task scheduler.

Not the best workaround, I know, but it saves you the pain of doing this all the time.

Alternatively, if you have VMware software, you can just quit using Hyper-V as docker machine and use a VMware docker machine, I was able to get it working a few days ago. Actuality this is a issue because of issues on Hyper-V network connectivity.

**Comment by JackSlaterIV** [comment]
> 
> 
> JackSlaterIV, I actually have been able to make this workaround permanent, all it takes is defining a PowerShell script where you get all network interfaces and loop them setting as private. With this script, you can define a startup task on Windows task scheduler.
> 
> Not the best workaround, I know, but it saves you the pain of doing this all the time.
> 
> Alternatively, if you have VMware software, you can just quit using Hyper-V as docker machine and use a VMware docker machine, I was able to get it working a few days ago. Actuality this is a issue because of issues on Hyper-V network connectivity.

Thanks man. I had followed your tip and that seemed to work, but I still get the issue. For me it happens again in a matter of minutes. I hope the team will find a proper way to solve this.
At this point to fix this I need to uncheck and recheck the drive letter in settings->shared drives (no need to restart any container or authenticate again).

**Comment by brunolm** [comment]
```
    "knex": "^0.16.4-next2",
```

```bash
git clone git@github.com:brunolm/knex16bug.git
cd knex16bug
git checkout knex16
npm run api-i
docker-compose up
```

```
api_1    | > knex migrate:make --knexfile ./src/db/knexfile.ts "test"
api_1    |
api_1    | /usr/src/app/src/db/knexfile.ts:1
api_1    | (function (exports, require, module, __filename, __dirname) { import { database } from '../config'
api_1    |                                                               ^^^^^^
api_1    |
api_1    | SyntaxError: Unexpected token import
api_1    |     at createScript (vm.js:80:10)
api_1    |     at Object.runInThisContext (vm.js:139:10)
api_1    |     at Module._compile (module.js:617:28)
api_1    |     at Object.Module._extensions..js (module.js:664:10)
api_1    |     at Module.load (module.js:566:32)
api_1    |     at tryModuleLoad (module.js:506:12)
api_1    |     at Function.Module._load (module.js:498:3)
api_1    |     at Module.require (module.js:597:17)
api_1    |     at require (internal/module.js:11:18)
api_1    |     at initKnex (/usr/src/app/node_modules/knex/bin/cli.js:62:25)
api_1    |     at Command.commander.command.description.option.action (/usr/src/app/node_modules/knex/bin/cli.js:172:24)
api_1    |     at Command.listener (/usr/src/app/node_modules/commander/index.js:315:8)
api_1    |     at emitTwo (events.js:126:13)
api_1    |     at Command.emit (events.js:214:7)
api_1    |     at Command.parseArgs (/usr/src/app/node_modules/commander/index.js:654:12)
api_1    |     at Command.parse (/usr/src/app/node_modules/commander/index.js:474:21)
api_1    | npm ERR! code ELIFECYCLE
api_1    | npm ERR! errno 1
api_1    | npm ERR! knex16@1.0.0 migrate-make: `knex migrate:make --knexfile ./src/db/knexfile.ts "test"`
api_1    | npm ERR! Exit status 1
api_1    | npm ERR!
api_1    | npm ERR! Failed at the knex16@1.0.0 migrate-make script.
api_1    | npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
api_1    |
api_1    | npm ERR! A complete log of this run can be found in:
api_1    | npm ERR!     /root/.npm/_logs/2019-03-18T21_39_10_408Z-debug.log
```

**Comment by svetkomir** [comment]
For me this is strictly related to VPN. 

When I'm not connected to VPN it works fine. Then I connect and I start getting the error immediately. 

**What I did is I restarted Docker after connecting to VPN.**

**Comment by deanmcginndm** [comment]
> I had the same issue. I was able to resolve it by running:
> docker volume rm -f [name of docker volume with error]
> Then restarting docker, and running:
> docker-compose up -d --build
> 
> I tried these same steps without restarting my docker, but restarting my computer and that didn't resolve the issue. What resolved the issue for me was removing the volume with the error, restarting my docker, then doing a build again.

Unsure if you did, but please can you kindly explain the meaning of the "-d" and "--build" in "docker-compose up -d --build" please?

I don't mean "this is what the man-page says", rather I mean what is the human explanation of the flags?

**Comment by JackSlaterIV** [comment]
The only solution that works for me is this guide https://web.archive.org/web/20180719031325/http://peterjohnlightfoot.com/docker-for-windows-on-hyper-v-fix-the-host-volume-sharing-issue/
It has been reported in other replies, and I would give it a try. 
Basically you need to create a new user under Windows 10 then use it in Docker: something like 10.0.75.1\DockerUser with a password that does not contain special characters (which are the cause of the issue).

**Comment by svetkomir** [comment]
Adding this route after connecting to the VPN fixed it consistently for me. This is the default network configuration for Docker for Windows.

`route ADD 10.0.75.0 MASK 255.255.255.0 10.0.75.1`

**Comment by abarke** [comment]
**GlobalProtect VPN Client** and the **Cisco Anyconnect VPN Client** are the problems on our end. 
These interfere with mounting shared folders. 

Is there a solution for the VPN use case?

**Comment by rakekris** [comment]
> On Windows this may be due to a user password change. Uncheck the box to stop sharing the drive and then allow Docker to detect that you are trying to mount the drive and share it.
> ![image](https://user-images.githubusercontent.com/470472/36745728-49101172-1ba5-11e8-94da-1386cfba9702.png)

this worked for me. thank you

**Comment by nea** [comment]
Hi everyone

Does anybody has a different suggestion on how to solve this?

I restarted Docker, reset my credentials, created a special 'docker' user as admin on Windows, removed volumes, rebuild... but absolutely nothing is working.

I just have two simple bindings but it just will not work, no matter how many times I restart, reset or whatever.

```
volumes:
      - ./:/app
      - ./.docker/site.conf:/etc/nginx/conf.d/default.conf
```

If there would at least be some hint where to look, unfortunately it becomes frustrating...


Any help is highly appreciated.

Thanks a lot

**Comment by oldium** [comment]
If you do not insist on using stable software, there is a experimental [WSL2](https://devblogs.microsoft.com/commandline/wsl-2-is-now-available-in-windows-insiders/) way of running Docker directly on Windows (see [example](https://devblogs.microsoft.com/commandline/announcing-wsl-2/)) in virtualized container (plus some manual [port forwarding](https://github.com/microsoft/WSL/issues/4150) is needed to access containers from other computers) - in July the Docker folks plan to release experimental [Docker with WSL2 support](https://engineering.docker.com/2019/06/docker-hearts-wsl-2/). Another way is to use experimental LCOW, but latest LCOW kernel image [does not boot](https://github.com/linuxkit/lcow/issues/47) and my personal experience is that containers are not able to start after reboot - I have to rebuild them on every reboot (which does not happen often). Otherwise LCOW is stable for me, I have not seen `mkdir` error for ages. I look forward to test WSL2 way as soon as experimental Docker with WSL2 support is out.

**Comment by nea** [comment]
Hi @oldium 

I am also looking forward to a finalized WSL2 integration. The collaboration between Microsoft and Docker seems promising for quite some time now and will (hopefully) align development needs throughout operatings systems in the future.

For now: I just tried Docker Edge Channel because what did I have to loose ^^' And it works! Docker Edge 2.0.5.0 works. Docker 2.0.0.4 (I think) did not work. Same `docker-compose.yml` file.


Thanks

**Comment by HakaiNoKami** [comment]
A solução de @rfulwell funcionou para mim. Obrigado!

**Comment by Laubeee** [comment]
`docker-compose up -V` worked for me.  

from `docker-compose up --help`:
```
-V, --renew-anon-volumes   Recreate anonymous volumes instead of retrieving data from the previous containers.
```

**Comment by mapruyas** [comment]
For those who got here, try above solutions and didnt work... I was mounting volumens that were symbolic links, some of them worked, others where erroring with file exists. 

The problem was the way I created the links, creating the links this way solve the issue for me.

mklink /J "C:\path_to_target\" "C:\path_to_link"

then it worked

**Comment by JoeSchr** [comment]
> `docker-compose up -V` worked for me.
> 
> from `docker-compose up --help`:
> 
> ```
> -V, --renew-anon-volumes   Recreate anonymous volumes instead of retrieving data from the previous containers.
> ```

That worked for me. quick and easy! Thx

**Comment by joshlsullivan** [comment]
I just ran `docker-compose down` and then `docker-compose up`. Worked for me.

**Comment by glorynguyen** [comment]
I have tried with `docker container prune` then press `y` to remove all stopped containers. This issue has gone.

**Comment by EliCDavis** [comment]
Running `docker-compose down` did not solve the issue. I didn't have to reset drive credentials. I just reset the docker desktop and it started working.

It's super annoying trying to get people to adopt docker where I'm at, as we're routinely running into issues on our dev machines, and almost always the fix is "just restart docker". It fixes issues like this one and others. It's almost every day, on every computer.

**Comment by psyclr** [comment]
![image](https://user-images.githubusercontent.com/8769980/65491646-26558b80-deb8-11e9-8631-44066176c358.png)
As mentioned, it should be checked. But even if it already checked - uncheck and check again, donno why, but works

**Comment by joguarino** [comment]
I don't know if it has been already told, but I solved this problem simply setting more privileges to the execution of Docker Desktop (I'm on Windows 10).

First of all, quit Docker Desktop if running. Then go to the installation folder (C:\Program Files\Docker\Docker), and right-click "Docker Desktop.exe" to open the contextual menu, where you have to click "Program Compatibility Troubleshooter".
In the opened window click on "Troubleshoot program",
then check "The program requires additional permissions": this will give administrator permission to the program.
In the next window just click on "Test the program" and wait for Docker Desktop to start: check the mount problem is solved and, if it is so, go to the next window and click on "Yes, save these settings for this program".

It should work.

**Comment by centur** [comment]
another option that helped me - reset docker to factory defaults.

**Comment by wussup** [comment]
In my case, the problem was with Windows user credentials - my password was expired. Relogin helped.

**Comment by xer10m** [comment]
I have the following problem:
We use Docker for our CI / CD to perform integration tests.
Our build server has 10 agents. Each of these agents runs Docker Desktop (Windows machines).
About once a day we get an "ERROR: for compose_db_1 Can not start service db: error while creating mount source path: mkdir / host_mnt / c: file exists" or an „ERROR: for compose_db_1 Cannot create container for service db: b'Drive sharing failed for an unknown reason'“ error.
But it is completely random when and on which agent the error occurs.
Once an agent throws an error, we need to restart the build and it will run again. It does not matter whether the build is triggered on the same machine.
Does anyone have an idea what we can do?

**Comment by EliCDavis** [comment]
@xer10m this is a shot in the dark, but maybe you could try doing a docker desktop reset every time before you try running your tests. I have no idea whether or not that'll work though

**Comment by pavan2106** [comment]
> @rfulwell's solution works! But it's easier to use the Reset Credentials... button.
> 
> ![image](https://user-images.githubusercontent.com/33459309/38609289-96062c54-3d75-11e8-936d-ae59ff07811c.png)

Your suggestion helped me for the day...  Thank you very much..

**Comment by abarke** [comment]
Be aware, every time your windows password changes. You will need to click "Reset credentials..."

**Comment by oldium** [comment]
Sometimes this helps even without changing the password...

**Comment by gaborari** [comment]
I ran into the same issue when using the following command:
_docker run --privileged --rm -it -v /c/{hostFolder}/:/{targetFolder}/ --name {name} {imageName}_
Error response from daemon: error while creating mount source path...

After several credential resets, almost gave up, then I've added the computername before /c/:
_docker run --privileged --rm -it -v /**{computerName}**/c/{hostFolder}/:/{targetFolder}/ --name {name} {imageName}_

The error is gone now, hope it helps.

**Comment by PauloPortfolio** [comment]
Step 01: [https://tomssl.com/2018/01/11/sharing-your-c-drive-with-docker-for-windows-when-using-azure-active-directory-azuread-aad/](https://tomssl.com/2018/01/11/sharing-your-c-drive-with-docker-for-windows-when-using-azure-active-directory-azuread-aad/)

Step 02: @alexkoepke Solution. [https://github.com/docker/for-win/issues/3385](https://github.com/docker/for-win/issues/3385)

**Comment by UnnitMetaliya** [comment]
Just restart your system and it will ask for your system password. Give your password to Docker as it needs to access your file system. 

Every once in a while, you need to make sure your Docker Desktop is running.

**Comment by nidhimundra** [comment]
I ran `docker volume prune`, and restarted my docker and it worked.

**Comment by rm-bergmann** [comment]
I am getting this problem now. Windows 10 upgraded to Build 19013.vb_release.191025-1609 (Insiders build) and now I can't start my docker container (I have tried everything listed above.)

ERROR: for [container-name]  Cannot start service [container-name]: error while creating mount source path '/host_mnt/c/www/docker-projects/cdn': mkdir /host_mnt/c: file exists

1 thing that looks different is `host_mnt`, in WSL1 if I `ls` the root directory, it's just 'mnt' so not sure if something has changed or if it was like that before.

I can't seem to roll back the update, so I will try to use WSL2.

**Comment by coder82** [comment]
@rfulwell worked for me too

**Comment by oldium** [comment]
Just tested latest [WSL2 Backend](https://engineering.docker.com/2019/10/new-docker-desktop-wsl2-backend/) (no need to have Ubuntu inside WSL2 any more) with Windows build from Insider's Slow Ring (build [19013.1122](https://blogs.windows.com/windowsexperience/2019/10/29/announcing-windows-10-insider-preview-build-19013/)) with Bitwarden Docker image and it works very well - fast startup (after first setup, of course), no errors shown during containers setup, running smoothly. Well done! This milestone should mark the end of this issue to everyone (I hope, fingers crossed).

**Comment by rm-bergmann** [comment]
Installing WSL 2 solves my problem, and is a huge improvement over all.

**Comment by oldium** [comment]
Just info for others planning to try WSL2 - latest Docker Desktop Edge [2.1.6.0](https://docs.docker.com/docker-for-windows/edge-release-notes/) now requires build [19018](https://docs.microsoft.com/en-us/windows-insider/flight-hub/), which is in Fast Ring. Slow Ring is not sufficient anymore.

**Comment by nebuk89** [comment]
Outside of WSL2 we have also now moved over from using Samba to using GRPCFuse in 2.1.6.0, it would be great if people with this issue could let me know if they still experience it on 2.1.6 outside of WSL.

**Comment by lrpinto** [comment]
I am not sure what worked for me, but resetting credentials seems to have led me into the right direction, as after this my error changed to a firewall issue.

I followed instructions here https://docs.docker.com/docker-for-windows/#firewall-rules-for-shared-drives. And before rebuilding I have also pruned containers, images and volumes.

**Comment by flatline-studios** [comment]
I'd just like to bring it to people's attention, that there are issues surrounding Windows Juntions (and possibly symbolic links [symlinks]), with WSL 2.

```
volumes:
  - "C:\\Development\\Docker\\Privacy\\socks-my-vpn\\conf:/vpn:ro"
```

Works just fine on Docker for Windows originally. Then yesterday I made the switch to WSL 2... And my compose file was failing with a rather cryptic error:

```
docker container run -it -v "C:\\Development\\Docker\\Privacy\\socks-my-vpn\\conf:/vpn" ohpe/socks-my-vpn /bin/sh
docker: Error response from daemon: error while creating mount source path '/run/desktop/mnt/host/c/Development/Docker/Privacy/socks-my-vpn/conf': mkdir /run/desktop/mnt/host/c/Development: file exists.
```

I spent about an hour trying various things (this is the first container I was trying to get running on WSL 2). I couldn't work out for the life of me why it was attempting to create the `C:\Development` folder... Then it dawned on me, `C:\Development` is a **junction** that points to `S:\Development`.

I'm not quite sure what's going on behind the scenes, as Docker knows that `C:\Development` exists (as the rather cryptic error suggests), but is unable to actually access it..?

Anyway, changing the drive letter in the volume bind mount to be:

```
volumes:
  - "S:\\Development\\Docker\\Privacy\\socks-my-vpn\\conf:/vpn:ro"
```

Solved the issue, and allowed the container to run correctly.

I'm mentioning this here to save someone else some time (I realise this is pretty niche), though, I'm still not sure why that happened in the first place?

Could the error message be made a little more explicit if it encounters a junction/symlink?

**Comment by artemmolotov** [comment]
@nebuk89
> Outside of WSL2 we have also now moved over from using Samba to using GRPCFuse in 2.1.6.0, it would be great if people with this issue could let me know if they still experience it on 2.1.6 outside of WSL.

After upgrading to 2.1.7 and a number of small manual tests, I did not notice this problem (no longer appeared).

But in version 2.1.7, I have a problem with changing mounted files (as if they are in read-only mode).

**Comment by djs55** [comment]
@artemmolotov thanks for the report! Regarding the problem
> After changing a file on the host, the file does not change in the container...

I've fixed a number of bugs related to this, would you like to try an experimental build to see if it helps?

The build is https://download-stage.docker.com/win/edge/41944/Docker%20Desktop%20Installer.exe

If it still reproduces could you
-  reproduce the problem i.e. change a file on the host, wait a second or two, read the file in the container (with the old contents) and then upload a diagnostic report and quote the ID? The logs will contain useful diagnostic traces
- could you share reproduction instructions? Ideally an example `docker run -v` or a `docker-compose.yml` so that I could try to reproduce the problem myself.

There's also a diagnostic tool which you can play with. The command
```
docker run -v /run:/run -it djs55/grpcfuse-control:2.2.0 statistics calls --follow
```
should show a live trace of FUSE filesystem events: both cache invalidations and reads/writes.

The command
```
docker run -v /run:/run -it djs55/grpcfuse-control:2.2.0 invalidate all
```
should invalidate all cache entries -- this ought to temporarily "fix" the problem (at least it would be good to verify that it actually does)

**Comment by artemmolotov** [comment]
@djs55 
Yes, I found the #5449 and tried the build (41944) 3 days ago. I checked the version of the build before the tests. The error was present. 

Now I don't see the error, although the code and configuration haven't been modified. I think this could be resolved due to a reboot of the OS, Docker Desktop, the cache invalidation or something like that.

If I reproduce the error in the future, I'll write in #5449 or a new issue.

Thank you. :)

**Comment by FritsvanCampen** [comment]
Like @flatline-studios mentioned I can confirm that Docker build `19.03.5, build 633a0ea` no longer works with junctions. It worked fine before; previous build was maybe 2 months old?

Changing the volume mount to the non-junctioned path solved the issue for me. Would be nice if the old behavior could be restored and/or the error message be made less cryptic.

**Comment by peekepic** [comment]
Hi all, i reproduce same issue and i success to understand my problem.
Windows 10 Pro.

Inside the windows folder that i tried to share, i had a virtual windows link to another directory make with the "MKLINK" previously. If i remove this virtual link, the issue disappear.

Hope this trick helps you.

**Comment by oomkarpanditstudent** [comment]
Updated to latest docker desktop 2.2.0 on 21st Jan 2020. Had issues "Errror mkdir /host_mnt/c: file exists when restarting docker container with mount"

I tried above all without luck then moved the folder I was mounting under C:\users and it worked. 

Lost whole day, hope others can try this fix and save time.

**Comment by alexey-igrychev** [comment]
In my case, docker has stopped supporting short windows filenames... I face it using the CLI tool written in Go that uses temp directory, https://golang.org/pkg/io/ioutil/#TempDir, as container mount.

To put it more simply:
```
$ docker run -ti --rm -v "C:\Users\ADMINI~1\AppData\Local\Temp\2":/host alpine
C:/Program Files/Docker/Docker/resources/bin/docker.exe: Error response from daemon: error while creating mount source path '/host_mnt/c/Users/ADMINI~1/AppData/Local/Temp/2': mkdir /host_mnt/c/Users/ADMINI~1: file exists.

$ docker run -ti --rm -v "C:\Users\Administrator\AppData\Local\Temp\2":/host alpine
/ # ok
```

```
$ docker info
Client:
 Debug Mode: false

Server:
 Containers: 1
  Running: 0
  Paused: 0
  Stopped: 1
 Images: 4
 Server Version: 19.03.5
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: b34a5c8af56e510852c35414db4c1f4fa6172339
 runc version: 3e425f80a8c931f88e6d94a8c831b9d5aa481657
 init version: fec3683
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 4.19.76-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 2
 Total Memory: 1.92GiB
 Name: docker-desktop
 ID: DNFH:TI6L:TVPB:Z7ZK:IK65:6V57:IMZY:FZZL:XEGE:C3QF:KQ3Q:CYPZ
 Docker Root Dir: /var/lib/docker
 Debug Mode: true
  File Descriptors: 35
  Goroutines: 52
  System Time: 2020-01-22T20:17:52.675875272Z
  EventsListeners: 3
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine
```

**Comment by AndrewSav** [comment]
In my case I'm getting this error because I have a junction in the host path

**Comment by f2calv** [comment]
After upgrading to Docker Desktop 2.2 I had the same issue as above;
> ERROR: for redis  Cannot start service redis: error while creating mount source path '/host_mnt/c/Users/XXXX/source/repos/MyProject/_DockerCompose/redis': mkdir /host_mnt/c/Users/XXXX/source/repos/MyProject: file exists

Below is a snippet from my failing compose file, the docker-compose.yaml file is located at c:/Users/XXXX/source/repos/MyProject/docker-compose.yaml on Windows 10.

```yaml
version: '3.7'

services:
  redis:
    container_name: redis
    image: "redis:alpine"
    ports:
      - "6379:6379"
    volumes:
      - ./_DockerCompose/redis:/data #fails with docker desktop 2.2
      #- c:/temp/_DockerCompose/redis:/data #this (obviously?) works with docker desktop 2.2.0.0, was working with previous versions
    entrypoint: redis-server --appendonly yes

  p3x:
    container_name: redisui
    image: "patrikx3/p3x-redis-ui"
    ports:
      - "7843:7843"
    volumes:
      - ./_DockerCompose/p3x-redis-ui-settings:/settings #fails with docker desktop 2.2.0.0, was working with previous versions
      #- c:/temp/_DockerCompose/p3x-redis-ui-settings:/settings #this (obviously?) works with docker desktop 2.2
    depends_on:
      - redis
```
When switching the volume mounts to point to c:/temp the problem goes away, so yes something fundamental has changed or there is a junction issue?

**Comment by PetaPetaPeta** [comment]
Upgrading to Docker Desktop 2.2 resulted in the same error for me. I solved it by changing my relative paths in volumes to full paths. 
`./project1`
to 
`c:/Users/docker/Documents/code/project1`

**Comment by mat007** [comment]
The apparently similar error your seeing in 2.2.0.0 has a different root cause, the related ticket is #5516

**Comment by redec** [comment]
FYI I just installed 2.2 and encountered this same error when trying to run any container with a mounted volume.  In my case it was because I had my host directory casing incorrect (c:\**u**sers\whatever instead of c:\**U**sers\whatever).  I'm just mentioning it in case it helps someone else in the same situation.

**Comment by EOussama** [comment]
Same here after the 2.2 update.

**Comment by lvkins** [comment]
I'm facing the same problem right now.

Been trying latest Docker Desktop for Windows as well as the Docker Desktop from edge channel.

My setup details:

```
OS Name:                   Microsoft Windows 10 Pro
OS Version:                10.0.18363 N/A Build 18363
```

```
> docker version
Client: Docker Engine - Community
 Version:           19.03.5
 API version:       1.40
 Go version:        go1.12.12
 Git commit:        633a0ea
 Built:             Wed Nov 13 07:22:37 2019
 OS/Arch:           windows/amd64
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          19.03.5
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.12
  Git commit:       633a0ea
  Built:            Wed Nov 13 07:29:19 2019
  OS/Arch:          linux/amd64
  Experimental:     true
 containerd:
  Version:          v1.2.10
  GitCommit:        b34a5c8af56e510852c35414db4c1f4fa6172339
 runc:
  Version:          1.0.0-rc8+dev
  GitCommit:        3e425f80a8c931f88e6d94a8c831b9d5aa481657
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

```
> docker-compose version
docker-compose version 1.25.1-rc1, build d92e9bee
docker-py version: 4.1.0
CPython version: 3.7.5
OpenSSL version: OpenSSL 1.1.1d  10 Sep 2019
```

When trying uncheck & check again shared drives in the Settings > Resources > File Sharing I'm being prompted to enter the user password all the time (I enter correct password). This happens only AFTER I face the `mkdir /host_mnt/c: file exists` error, when I restart Docker I can (un)check shared drives successfully again, but as soon as the error comes by, it's failing.

**Comment by PauloPortfolio** [comment]
I had a lot of problems with volumes. I catalog some solutions about theses troubles. I hope taht thses solutions can help someone in this topic.

- **Error 1: ** "Drive has not been shared":
-Solution: 
1.	In cmd with _**powershell admin rights**_ and enter the following commands:
> net user /add DockerDiskSharing DockerDiskSharing
> net localgroup Administrators DockerDiskSharing /add
> net localgroup docker-users DockerDiskSharing /add
> icacls "%UserProfile%" /q /c /t /grant docker-users:F

> 2.	In **Docker Daemon** > Settings -> Shared Drives screen:
•	click Reset credentials  >  tick drive C  >  click Apply
•	type DockerDiskSharing in user name and DockerDiskSharing in password
•	click OK

- **Error 2:** "error while creating mount  source path" or "drive has not been shared":
-Solution: 
> 1. Uncheck/remove the port(445), in the **firewall** rule. For example in kaspersky:
![image](https://user-images.githubusercontent.com/51207441/73139526-1a2b3780-402c-11ea-9725-ac60c2a3f4dd.png)
![image](https://user-images.githubusercontent.com/51207441/73139530-231c0900-402c-11ea-92e2-92784078e46e.png)
![image](https://user-images.githubusercontent.com/51207441/73139532-27e0bd00-402c-11ea-865c-eb91f0d05e20.png)
![image](https://user-images.githubusercontent.com/51207441/73139550-5068b700-402c-11ea-9846-9b3c8221f810.png)

>2.	No Docker Daemon > Settings -> Shared Drives screen:
•	click Reset credentials  >  tick drive C  >  click Apply
•	type DockerDiskSharing in user name and DockerDiskSharing in password
•	click OK

- Error 03:  “permission denied” in the docker-compose, dockerfile ou simple run
-Solution: 
1. Give **FULL-PERMISSION** for the **HOST-FOLDER** 
<img width="743" alt="Untitled1" src="https://user-images.githubusercontent.com/51207441/73139667-89edf200-402d-11ea-9e0c-b99344386451.png">
<img width="242" alt="Untitled2" src="https://user-images.githubusercontent.com/51207441/73139669-8ce8e280-402d-11ea-9a04-dfc0ba69dfc3.png">

Ihope that these tips help someone

**Comment by dejanmr** [comment]
I can also confirm the issue, present after upgrade to 2.2.0.0 (42247),engine 19.03.5 on windows 10 Professional.
In my case issue is present because of symbolic links.
It was working fine for years, until now, I was always up-to-date with the latest update.
I need symbolic links it do avoid issues I have with folders with space in name, which I can't avoid due to some legacy software.

**Comment by kevingentile** [comment]
Docker Desktop 2.1.7.0 (41536) on Windows 10

The error I get is as follows: 
```
ERROR: for container_alpine_1  Cannot start service alpine: error while creating mount source path '/host_mnt/c/Users/[...]/dist/alpine': mkdir /host_mnt/c/Users/[...]/dist/alpine: file exists
```

In `docker-compose.yaml` I have a mount:
```
volumes:
    - "./dist/alpine:/var/tmp/dist"
```

On the first run, the container creates the `dist` folder on my local machine. I hit the error when I delete the `dist` directory on my local machine and re-run the container. 

I would expect docker to re-create `dist` again with `docker-compose up`.

To mitigate this issue I can restart Docker Desktop from the task bar, and the container executes as expected.

**Comment by cliffchapmanrbx** [comment]
After upgrading to 2.2 I'm also getting the same error message, this time with a docker-compose file. I'm using relative paths to locate a configuration file and mount it.

Original docker-compose.yaml that was working immediately prior to the update:
```yaml
        volumes:
            - "./nats-server.conf:/etc/nats/nats-server.conf:ro"
```

"Fixed" configuration that is now working presently:
```yaml
        volumes:
            - "C:/Git/nats-cluster/spout/nats-server.conf:/etc/nats/nats-server.conf:ro"
```

Observation: The "Git" folder is capitalized on my windows system. When I use the relative path, the error message has it lowercased:
```
error while creating mount source path '/host_mnt/c/git/nats-cluster/spout/nats-server.conf': mkdir /host_mnt/c/git: file exists
```

When I explicitly uppercase it in the configuration it's fine. *Are relative paths broken because Docker is tolower()'ing something it shouldn't be?*

**Comment by mj2015** [comment]
Okay, just been through this, and the last comment triggered my solution. I had 
      -v c:\data\:/PersistentData
and it was failing with this mount error. Nothing special about my C:\Data folder at all. Hmm, tried 
      -v c:\Data:/PersistentData
and it works! Ouch! Hope this helps someone else.

**Comment by jsalvadorcaffarena** [comment]
> 
> 
> Okay, just been through this, and the last comment triggered my solution. I had
> -v c:\data:/PersistentData
> and it was failing with this mount error. Nothing special about my C:\Data folder at all. Hmm, tried
> -v c:\Data:/PersistentData
> and it works! Ouch! Hope this helps someone else.

Just to add that this also helped in my case after getting these errors suddenly after upgrading Docker. There seems to have been a regression to case sensitive paths for Windows.

**Comment by cliffchapmanrbx** [comment]
Looks like the issue with version 2.2.0.0 is being tracked in #5516, we should probably not continue that discussion in this thread here.

**Comment by afilcheff** [comment]
> 
> 
> After upgrading to 2.2 I'm also getting the same error message, this time with a docker-compose file. I'm using relative paths to locate a configuration file and mount it.
> 
> Original docker-compose.yaml that was working immediately prior to the update:
> 
> ```yaml
>         volumes:
>             - "./nats-server.conf:/etc/nats/nats-server.conf:ro"
> ```
> 
> "Fixed" configuration that is now working presently:
> 
> ```yaml
>         volumes:
>             - "C:/Git/nats-cluster/spout/nats-server.conf:/etc/nats/nats-server.conf:ro"
> ```
> 
> Observation: The "Git" folder is capitalized on my windows system. When I use the relative path, the error message has it lowercased:
> 
> ```
> error while creating mount source path '/host_mnt/c/git/nats-cluster/spout/nats-server.conf': mkdir /host_mnt/c/git: file exists
> ```
> 
> When I explicitly uppercase it in the configuration it's fine. _Are relative paths broken because Docker is tolower()'ing something it shouldn't be?_

Setting the path explicitly worked out for me, thanks

**Comment by EOussama** [comment]
Simply lowercasing every folder in my driver that matches the volume path seems to solve this for me. I hope a fix for this is introduced that doesn't force me to change anything.

**Comment by noobhacker** [comment]
I clone a project from Github using Visual Studio. The IDE automatically creates a directory under C:\Users\MyName\source\repo
However, for some reason, the docker is trying to mount to
C:\Users\MyName\Source\Repo
Rename source and repo to Source and Repo solved the problem.

**Comment by chopnut** [comment]
This is happening to me as well with the latest version.
Every time I load my wordpress docker-compose, it would mount 1 file from the wp container. And the container would stop, and when I restart docker-compose it will flag an error. About unable to mount.

This has never happened to me before.

**Comment by stephen-turner** [comment]
I think this ticket has accumulated more than one bug. As far as we know, the original issue (and most or all of the later issues) are fixed in 2.2.0.3, so I'm going to close this ticket. If there are further issues in the latest version, please open a new ticket. Thank you.

**Comment by ntnhan0612** [comment]
Login Docker again solved the issue. It worked for me !

**Comment by adamb70** [comment]
Restarting docker after deleting the container and volume worked for me.

**Comment by rustyx** [comment]
This is still happening randomly on latest Docker for Windows running on Windows 10 x64.

Simply un-sharing and re-sharing the C drive in Docker settings "fixes" the issue, without Docker restart.

So the underlying issue seems to still be there.
